discord/sink.py

```diff
@@ -0,0 +1,775 @@
+        raise NotImplementedError()
+
+    def _get_new_path(self, path, ext, new_name=None):
+        ext = "."+ext
+        directory, name = os.path.split(path)
+        name = new_name + ext if new_name is not None else ".".join(name.split(".")[:-1]) + ext
+        return os.path.join(directory, name)
+
+
+class WaveAudioFileSink(AudioFileSink):
+    """Extends :class:`AudioFileSink` and defines a convert_file method."""
+    CHUNK_WRITE_SIZE = 64
+
+    def convert_file(self, file, new_name=None):
+        """Takes a file object that contains raw audio data and writes
+        it to a wave audio file.
+
+        Parameters
+        ----------
+        file: :class:`IO`
+            File object that contains raw audio data
+        new_name: Optional[:class:`str`]
+            Name for the wave file excluding ".wav". Defaults to current name if None.
+
+        Returns
+        -------
+        :class:`str`
+            Path to the new audio file
+        """
+        path = self._get_new_path(file.name, "wav", new_name)
+        with wave.open(path, "wb") as wavf:
+            wavf.setnchannels(OpusDecoder.CHANNELS)
+            wavf.setsampwidth(OpusDecoder.SAMPLE_SIZE // OpusDecoder.CHANNELS)
+            wavf.setframerate(OpusDecoder.SAMPLING_RATE)
+            while frames := file.read(OpusDecoder.FRAME_SIZE*self.CHUNK_WRITE_SIZE):
+                wavf.writeframes(frames)
+        return path
+
+
+class MP3AudioFileSink(AudioFileSink):
+    """Extends :class:`AudioFileSink` and defines a convert_file method."""
+
+    def convert_file(self, file, new_name=None):
+        """Takes a file object that contains raw audio data and writes
+        it to a mp3 audio file.
+
+        Parameters
+        ----------
+        file: :class:`IO`
+            File object that contains raw audio data
+        new_name: Optional[:class:`str`]
+            Name for the wave file excluding ".mp3". Defaults to current name if None.
+
+        Returns
+        -------
+        :class:`str`
+            Path to the new audio file
+        """
+        path = self._get_new_path(file.name, "mp3", new_name)
+        args = ['ffmpeg', '-f', 's16le', '-ar', str(OpusDecoder.SAMPLING_RATE),
+                '-ac', str(OpusDecoder.CHANNELS), '-y', '-i', file.name, path]
+        try:
+            process = subprocess.Popen(args, creationflags=subprocess.CREATE_NO_WINDOW)
+        except FileNotFoundError:
+            raise ClientException('ffmpeg was not found.') from None
+        except subprocess.SubprocessError as exc:
+            raise ClientException('Popen failed: {0.__class__.__name__}: {0}'.format(exc)) from exc
+        process.wait()
+
+
+class AudioReceiver(threading.Thread):
+    def __init__(
+        self,
+        sink: AudioSink,
+        client: 'VoiceClient',
+        *,
+        decode: bool = True,
+        after: Optional[Callable[[AudioSink, Optional[Exception]], Any]] = None,
+    ) -> None:
+        threading.Thread.__init__(self)
```

Comment:
This is an old backport style call to `super().__init__()`, you're free to switch to this.

Reply:
I used `thread.Thread` to mimic the `AudioPlayer` class, but I suppose that doesn't matter at all.